appcenter-analytics/Analytics.js

```diff
@@ -16,9 +16,31 @@ module.exports = {
     // async - returns a Promise
     setEnabled(enabled) {
         return AppCenterReactNativeAnalytics.setEnabled(enabled);
+    },
+
+    // async - returns a Promise
+    getTransmissionTarget(targetToken) {
```

Comment:
this should not be async

Reply:
Since all ReactNative bridge interactions are asynchronous, I don't understand how this can not be async?

Reply:
I didn't look like async in the resolve clause you just pass another call, is that async as well? if yes why do we have intermediate code.

Reply:
Forgot about that limitation in React, not ideal but no choice I guess.

Reply:
+1 for @bmourat, react native is asynchronous. Actually, there is an undocumented way to make blocking synchronous call via [RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD](https://github.com/facebook/react-native/blob/master/React/Base/RCTBridgeModule.h#L166-L183), but it's highly discouraged and can be problematic.